File: assignments/assignment-2-solution/src-csharp/TeaShop/Domain/PaymentStrategy/IPaymentStrategy.cs
using Assignment2Solution.Domain.Inventory;

namespace Assignment2Solution.Domain.PaymentStrategy;

/// <summary>
///     Defines a strategy for processing payments for inventory items.
/// </summary>
public interface IPaymentStrategy
{
    /// <summary>
    ///     Processes the checkout for a specific item and quantity.
    /// </summary>
    /// <param name="item">The inventory item being purchased.</param>
    /// <param name="quantity">The quantity being purchased.</param>
    /// <param name="output">The text writer for displaying checkout information.</param>
    void Checkout(InventoryItem item, int quantity, TextWriter output);
}

File: assignments/assignment-2-solution/src-java/src/main/java/assignment2solution/domain/payment/IPaymentStrategy.java
package assignment2solution.domain.payment;

import assignment2solution.domain.inventory.InventoryItem;

import java.io.Writer;

/**
 * Defines a strategy for processing payments for inventory items.
 */
public interface IPaymentStrategy {
    /**
     * Processes the checkout for a specific item and quantity.
     */
    void checkout(InventoryItem item, int quantity, Writer output);
}
